### DISCUSSION TITLE
[Unknown Mod][Secondary Skills] Changes to the Secondary Skills [Edited]

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2990] [Ideas]
**Changes to the Secondary Skills.**

**Diplomacy:**
With Diplomacy skill army camps will sometimes offer to join your army.
Percentage shown is the percentage of monsters in the stack.

Basic: 25% will join, cost 100%.
Advanced: 50% will join, cost 100%.
Expert: 100% will join, cost 100%.

Tweaks:
All the creatures will always join your army but you pay different cost for each Diplomacy level.

Basic: 100% will join, cost 100%.
Advanced: 100% will join, cost 75%.
Expert: 100% will join, cost 50%.

**Eagle Eye:**
Your hero gets a chance to learn spells that have been cast against him in combat.

Basic: 20% chance, 2nd level max.
Advanced: 30% chance, 3rd level max.
Expert: 40% chance, 4th level max.

Tweaks:
Your hero learns spells that have been cast against him in combat, and restores spell points each time the opponent hero casts a spell during combat.
The hero can restore spell points above his max spell points (ex: 22/20).
The hero can use the learned spells during the current combat and he can learn several spells during the same combat (a window is displayed during combat each time the hero learn a spell and it is also indicated in the information bar at the bottom of the screen).
The Dispel Magic spell animation can be used on the hero when spell points are restored.

Basic: 100% chance, 3rd level max, your hero restores 1 spell point multiplied by the level of the spell the opponent hero casts, each time the opponent hero casts a spell during combat.
Advanced: 100% chance, 4th level max, your hero restores 2 spell points multiplied by the level of the spell the opponent hero casts, each time the opponent hero casts a spell during combat.
Expert: 100% chance, 5th level max, your hero restores 3 spell points multiplied by the level of the spell the opponent hero casts, each time the opponent hero casts a spell during combat.

On the adventure map, the hero also learns spells that have been cast by opponents heroes located in his viewable area.

Basic: 100% chance, 3rd level max.
Advanced: 100% chance, 4th level max.
Expert: 100% chance, 5th level max.

So your hero has Expert Eagle Eye.
If a opponent hero is located in your hero viewable area and casts Town Gate spell, there are 100% change your hero learns Town Gate spell (a window is displayed at the beginning of the turn to inform the player).

**Pathfinding:**
Reduces the movement penalty on rough terrains (Beach, Cracked land, Snow, Swamp, Desert).

Basic: -25% movement penalty.
Advanced: -50% movement penalty.
Expert: -100% movement penalty.

Tweaks:
Reduces the movement penalty on rough terrains (Beach, Cracked land, Snow, Swamp, Desert), and increases land movement.

Basic: -25% movement penalty, +5% movement points.
Advanced: -50% movement penalty, +10% movement points.
Expert: -100% movement penalty, +15% movement points.

The Hero Movement should be calculated this way:
(((Base Hero Movement points (1500 points) - Creature Movement penalty) x Secondary skill bonus (Logitics + Pathfinding + Scouting)) + Artifact bonus)

**Scouting:**
Increases hero's viewable area on adventure map.

Basic: +1 square.
Advanced: +2 squares.
Expert: +3 squares.

Tweaks:
Increases hero's viewable area on adventure map, and increases land movement.

Basic: +1 square,  +5% movement points.
Advanced: +2 squares, +10% movement points.
Expert: +3 squares, +15% movement points.

The Hero Movement should be calculated this way:
(((Base Hero Movement points (1500 points) - Creature Movement penalty) x Secondary skill bonus (Logitics + Pathfinding + Scouting)) + Artifact bonus)

**Mysticism:**
Increases the number of spell points regenerated by the hero each turn.

Basic: 1 spell point per day (2 with the hero spell point restored every day).
Advanced: 2 spell points per day (3 with the hero spell point restored every day).
Expert: 3 spell points per day (4 with the hero spell point restored every day).

Tweaks:
Increases the number of spell points regenerated by the hero each turn.
The hero regenerates spell points at start of combat.

Basic: your hero regenerates 2 spell points per day on the adventure map (3 with the hero spell point restored every day), your hero regenerates up to 5 spell points at start of combat.
Advanced: your hero regenerates 4 spell points per day on the adventure map (5 with the hero spell point restored every day), your hero regenerates up to 10 spell points at start of combat.
Expert: your hero regenerates 6 spell points per day on the adventure map (7 with the hero spell point restored every day), your hero regenerates up to 15 spell points at start of combat.

The hero doesn't restore spell points above his max spell points (ex: max 20/20).

**Navigation:**
Hero gets an extra movement on water.

Basic: +50% movement.
Advanced: +100% movement.
Expert: +150% movement.

Tweaks:
Hero gets an extra movement on water and generates gold each turn (never enough gold in Heroes 2), and protects troops from whirlpools.

Basic: +50% movement on water, 125 gold per day, and protects troops from whirlpools.
Advanced: +100% movement on water, 250 gold per day, and protects troops from whirlpools.
Expert: +150% movement on water, 500 gold per day, and protects troops from whirlpools.

**Estates:**
Hero generates gold each turn.

Basic: produces 100 gold per turn.
Advanced: produces 250 gold per turn.
Expert: produces 500 gold per turn.

Tweaks to stay better than Navigation:

Basic: produces 250 gold per turn.
Advanced: produces 500 gold per turn.
Expert: produces 1000 gold per turn.

**Ballistics:**
During a siege, the catapult’s shots, damage, and accuracy are increased.

Basic: 1 shot + extra dam.
Advanced: 2 shots + extra dam.
Expert: 2 shots + max dam.

Tweaks:
During a siege, the catapult’s shots, damage, and accuracy are increased.
During a siege, the hero gets a temporary additional bonus to Attack and Defense (when the hero is the attacker or the defender).

Basic: 1 shot + extra dam, +1 Attack and +1 Defense during siege.
Advanced: 2 shots + extra dam, +2 Attack and +2 Defense during siege.
Expert: 2 shots + max dam, +3 Attack and +3 Defense during siege.

**Note: all of this could eventually change, especially the stated values.** :)

**Comment by 1GodRage** [comment]
I have an idea for you, maybe you won't like it... xD (sorry if so ^^')

**Eagle Eye :**
Basic: 40% chance, 2nd level max.
Advanced: 50% chance, 4rd level max.
Expert: 100% chance, 5th level max.
The Hero can not learn spells of level 3~5 in Magic guild.
The Hero an learn spells from level 1 and 2 in Magic guild.
Learning Eagle Eye will unlearn any Wisdom skill.
During battle, Less mana cost if casting the same spell that the opponent just cast.

**Reply by LeHerosInconnu** [reply]
Hello @1GodRage,

> I have an idea for you, maybe you won't like it... xD (sorry if so ^^')
> 
> **Eagle Eye :** Basic: 40% chance, 2nd level max. Advanced: 50% chance, 4rd level max. Expert: 100% chance, 5th level max. The Hero can not learn spells of level 3~5 in Magic guild. The Hero an learn spells from level 1 and 2 in Magic guild. Learning Eagle Eye will unlearn any Wisdom skill. During battle, Less mana cost if casting the same spell that the opponent just cast.

You don't have to be sorry, these threads are meant to discuss and exchange opinions and ideas. :)

In my mod, I try to make all secondary skills attractive and always useful throughout a game.

Your idea may be interesting in some cases, but I'm afraid that the fact that Eagle Eye removes the Wisdom skill (and also prevents learning spells from levels three to five) in your idea, makes this skill even less desirable for the majority of players (and that's even if the spells cost less spell points to cast) except maybe the players who exclusively play Might heroes and in that case why would they choose to take the Widsom skill for one of their hero in the first place.

**Reply by 1GodRage** [reply]
I agree, the idea won't mix well with others skills. xD
In fact that would be a new gameplay... If I continue in my idea, the eagle eye also will give you 10~30% chance to find an artefact after each battle... Also, why not:  to be able to learn "natural spells" ! Spell that unknown creatures will do during travelling, and, the Eagle Eyed Hero will learn it!!
Like the dissimulation spell: many creatures just hide themselfes when an army cross their home.
Some of them use a magic spell for hiding! Dissimulation! "At the end of your turn and until tomorrow, you are dissimulated, nobody can see you, nor attack you."
xD

Anyway, just sharing some ideas, no real deep-mind-learning-AI-thought in it. xD

**Reply by LeHerosInconnu** [reply]
Hello @1GodRage,

> I agree, the idea won't mix well with others skills. xD In fact that would be a new gameplay...

Yes, completely. :)

> If I continue in my idea, the eagle eye also will give you 10~30% chance to find an artefact after each battle...

Eagle Eye has more to do with magic than with the "quality" of vision.
And the player will feel a bit "forced" to do combat to get the maximum number of artifacts. :D

> Also, why not: to be able to learn "natural spells" ! Spell that unknown creatures will do during travelling, and, the Eagle Eyed Hero will learn it!! Like the dissimulation spell: many creatures just hide themselfes when an army cross their home. Some of them use a magic spell for hiding! Dissimulation! "At the end of your turn and until tomorrow, you are dissimulated, nobody can see you, nor attack you." xD

That would already require implementing all these things in the game.
And Eagle Eye should not become the must have secondary skill either. :)

> Anyway, just sharing some ideas, no real deep-mind-learning-AI-thought in it. xD

Yes, I see. :)

**Reply by lioror** [reply]
I think expert shouldn't be 100%, as there is nothing perfect.
maybe 70% or 80% up to level 5.

**Reply by Alucard648** [reply]
> I think expert shouldn't be 100%, as there is nothing perfect. maybe 70% or 80% up to level 5.

Never below 100%! Nothing more frustrating that miss sometimes the only opportunity to get vital spell (like DD, or Town Portal) just by a bad dice roll, oftenly decicive in the game. Especially if you invest 2-3 levels onto that skill.

**Reply by LeHerosInconnu** [reply]
Hello @lioror and @Alucard648,

> I think expert shouldn't be 100%, as there is nothing perfect. maybe 70% or 80% up to level 5.

The change is designed to avoid this "maybe, maybe not" effect, so that the benefit of each secondary skill is known and reliable.
With the change done to Eagle Eye, the effect is known and the player has to rely on strategy, not "luck".

> > I think expert shouldn't be 100%, as there is nothing perfect. maybe 70% or 80% up to level 5.
> 
> Never below 100%! Nothing more frustrating that miss sometimes the only opportunity to get vital spell (like DD, or Town Portal) just by a bad dice roll, oftenly decicive in the game. Especially if you invest 2-3 levels onto that skill.

"This mod will be primarily designed to have fun, reduce micromanagement and frustration, for single-player game mode."
So yes, frustration will be avoided.
